src/IO/WriteBufferFromPocoSocketChunked.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include <Common/logger_useful.h>
+#include <IO/WriteBufferFromPocoSocket.h>
+
+
+namespace DB
+{
+
+class WriteBufferFromPocoSocketChunked: public WriteBufferFromPocoSocket
+{
+public:
+    explicit WriteBufferFromPocoSocketChunked(Poco::Net::Socket & socket_, size_t buf_size = DBMS_DEFAULT_BUFFER_SIZE) : WriteBufferFromPocoSocket(socket_, buf_size), log(getLogger("Protocol")) {}
```

Comment:
Maybe just in case add a check that `buf_size` > 4 or introduce some minimum size here? Otherwise user can set `max_read_buffer_size=2` for example and it will break this write buffer